ui/controls.js

```diff
@@ -587,48 +587,59 @@ shaka.ui.Controls = class extends shaka.util.FakeEventTarget {
     return false;
   }
 
-  /** @export */
-  async toggleFullScreen() {
-    if (document.fullscreenEnabled) {
-      if (document.fullscreenElement) {
-        if (screen.orientation) {
-          screen.orientation.unlock();
+  /** @private */
+  async enterFullScreen_() {
+    try {
+      if (document.fullscreenEnabled) {
+        if (document.pictureInPictureElement) {
+          await document.exitPictureInPicture();
+        }
+        await this.videoContainer_.requestFullscreen({navigationUI: 'hide'});
+
+        if (this.config_.forceLandscapeOnFullscreen && screen.orientation) {
+          // Locking to 'landscape' should let it be either
+          // 'landscape-primary' or 'landscape-secondary' as appropriate.
+          await screen.orientation.lock('landscape');
         }
-        await document.exitFullscreen();
       } else {
-        // If we are in PiP mode, leave PiP mode first.
-        try {
-          if (document.pictureInPictureElement) {
-            await document.exitPictureInPicture();
-          }
-          await this.videoContainer_.requestFullscreen({navigationUI: 'hide'});
-          if (this.config_.forceLandscapeOnFullscreen && screen.orientation) {
-            try {
-              // Locking to 'landscape' should let it be either
-              // 'landscape-primary' or 'landscape-secondary' as appropriate.
-              await screen.orientation.lock('landscape');
-            } catch (error) {
-              // If screen.orientation.lock does not work on a device, it will
-              // be rejected with an error. Suppress that error.
-            }
-          }
-        } catch (error) {
-          this.dispatchEvent(new shaka.util.FakeEvent(
-              'error', (new Map()).set('detail', error)));
+        const video = /** @type {HTMLVideoElement} */(this.localVideo_);
+        if (video.webkitSupportsFullscreen) {
+          video.webkitEnterFullscreen();
         }
       }
+    } catch (error) {
+      this.dispatchEvent(new shaka.util.FakeEvent(
+          'error', (new Map()).set('detail', error)));
+      // Entering fullscreen can fail
+      // if the user didn't interacting with the video.
+      // be rejected with an error. Suppress that error.
```

Comment:
This comment doesn't look accurate.  It seems like we're dispatching our own wrapped version and not suppressing the original.